### PR TITLE
fix: Load plugin models in vscode extension

### PR DESCRIPTION
**Pull request by jiashengguo**

### CODE REVIEW

File: packages/schema/src/language-server/zmodel-workspace-manager.ts [hunk 1]
```diff
@@ -1,21 +1,115 @@
...
+        const documents: LangiumDocument[] = [];
+        const collector = (document: LangiumDocument) => {
+            documents.push(document);
+            if (!this.langiumDocuments.hasDocument(document.uri)) {
+                this.langiumDocuments.addDocument(document);
+            }
+        };
+        // Even though we don't await the initialization of the workspace manager,
+        // we can still assume that all library documents and file documents are loaded by the time we start building documents.
+        // The mutex prevents anything from performing a workspace build until we check the cancellation token
```
Comment (ymc9) [inline packages/schema/src/language-server/zmodel-workspace-manager.ts:42]: What does this "mutex" mean here?

File: packages/schema/src/language-server/zmodel-workspace-manager.ts [hunk 2]
```diff
@@ -1,21 +1,115 @@
...
+                            collector(document);
+                            console.log(`Adding plugin document from ${path}`);
+
+                            pluginModels.delete(plugin);
+                            // early exit if all plugins are loaded
+                            if (pluginModels.size === 0) {
+                                return;
+                            }
+                        } catch {
+                            //no-op
```
Comment (ymc9) [inline packages/schema/src/language-server/zmodel-workspace-manager.ts:106]: Shall we add a warning log here?

File: packages/schema/src/language-server/zmodel-workspace-manager.ts (hunk 1 above)
Comment (jiashengguo) [reply packages/schema/src/language-server/zmodel-workspace-manager.ts]: It's copied from the base class `DefaultWorkspaceManager`.  I think it means that the `initializeWorkspace` function itself is actually running in the `mutex.lock`.  Anyway, I think I will just remove it

File: packages/schema/src/language-server/zmodel-workspace-manager.ts (hunk 2 above)
Comment (jiashengguo) [reply packages/schema/src/language-server/zmodel-workspace-manager.ts]: There might be multiple node_modules locations,  the warning will be shown eventually after traversing all the folders but not found.

**Review by ymc9 — APPROVED** [review]

**Pull request merged**